apps/credential-showcase-api-server/src/__tests__/issuer.repository.test.ts

```
@@ -0,0 +1,251 @@
+      logo: unknownIconId,
+    };
+
+    await expect(repository.create(issuer)).rejects.toThrowError(`No asset found for id: ${unknownIconId}`)
+  })
+
+  it('Should get issuer by id from database', async (): Promise<void> => {
+    const issuer: NewIssuer = {
+      name: 'example_name',
+      type: IssuerType.ARIES,
+      credentialDefinitions: [credentialDefinition1.id, credentialDefinition2.id],
+      description: 'example_description',
+      organization: 'example_organization',
+      logo: asset.id,
+    };
+
+    const savedIssuer = await repository.create(issuer)
+    expect(savedIssuer).toBeDefined()
+
+    const fromDb = await repository.findById(savedIssuer.id)
+
+    expect(fromDb).toBeDefined()
+    expect(fromDb.name).toEqual(issuer.name)
+    expect(fromDb.type).toEqual(issuer.type)
+    expect(fromDb.description).toEqual(issuer.description)
+    expect(fromDb.organization).toEqual(issuer.organization);
+    expect(fromDb.credentialDefinitions.length).toEqual(2);
+    expect(fromDb.logo).not.toBeNull()
+    expect(fromDb.logo!.id).toBeDefined()
+    expect(fromDb.logo!.mediaType).toEqual(asset.mediaType)
+    expect(fromDb.logo!.fileName).toEqual(asset.fileName)
+    expect(fromDb.logo!.description).toEqual(asset.description)
+    expect(fromDb.logo!.content).toStrictEqual(asset.content)
+  })
+
+  it('Should get all issuers from database', async (): Promise<void> => {
+    const issuer: NewIssuer = {
+      name: 'example_name',
+      type: IssuerType.ARIES,
+      credentialDefinitions: [credentialDefinition1.id],
+      description: 'example_description',
+      organization: 'example_organization'
+    };
+
+    const savedIssuer1 = await repository.create(issuer)
+    expect(savedIssuer1).toBeDefined()
+
+    const savedIssuer2 = await repository.create(issuer)
+    expect(savedIssuer2).toBeDefined()
+
+    const fromDb = await repository.findAll()
+
+    expect(fromDb.length).toEqual(2)
+  })
+
```

??
Extra test "Should throw error when saving issuer with invalid credentialDefinitions" id I.E. credentialDefinitions: [12345],

---

added